preview/app.jsx

```diff
@@ -1,6 +1,25 @@
-import { hydrate } from 'react-dom';
-import { BrowserRouter } from 'react-router-dom';
-
+import { createRoot, hydrateRoot } from 'react-dom/client';
+import { BrowserRouter, HashRouter } from 'react-router-dom';
 import createRouter from './router';
 
-hydrate(<BrowserRouter>{createRouter()}</BrowserRouter>, document.getElementById('main'));
+
+const mainNode = document.getElementById('main');
+const isPrerender = Boolean(mainNode);
+
+if (isPrerender) {
+  hydrateRoot(mainNode, (
+    <BrowserRouter>
+      {createRouter()}
+    </BrowserRouter>
+  ));
+} else {
+  const node = document.createElement('main');
+  document.body.appendChild(node);
+
+  createRoot(node).render((
+    // eslint-disable-next-line no-undef
+    <HashRouter basename={PUBLIC_PATH}>
```

Comment:
don't know how it is resolved in running system (maybe just always undefined) but preview is running on my machine with latest hocs package (via yalc).

Reply:
Good catch, was a copy paste issue, I believe we don't need that part here